src/ctapipe/calib/camera/outlier.py

```diff
@@ -0,0 +1,112 @@
+"""
+Outlier detection algorithms to identify faulty pixels
+"""
+
+__all__ = [
+    "OutlierDetection",
+    "RangeBasedOutlierDetection",
+    "MedianBasedOutlierDetection",
+    "StdBasedOutlierDetection",
+]
+
+from abc import abstractmethod
+
+import numpy as np
+
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import List
+
+
+class OutlierDetection(TelescopeComponent):
+    """
+    Base class for outlier detection algorithms.
+    """
+
+    outliers_interval = List(
+        [-1.0, 1.0],
+        help=(
+            "Interval of the multiplicative factor for detecting outliers based on"
+            "the subcomponents."
+        ),
+    ).tag(config=True)
+
+    @abstractmethod
+    def __call__(self, column) -> bool:
+        """
+        Detect outliers in the provided table based on the specified column.
+
+        This function should be implemented by subclasses to define the specific
+        outlier detection approach. The function examines the statistics in the
+        given column of the table and returns a boolean mask indicating which
+        entries are considered outliers.
+
+        Parameters
+        ----------
+        column : astropy.table.Column
+            column with chunk-wise extracted statistics (mean, median, or std)
+            of shape (n_entries, n_channels, n_pix)
+
+        Returns
+        -------
+        boolean mask
+            mask of outliers of shape (n_entries, n_channels, n_pix)
+        """
+        pass
+
+
+class RangeBasedOutlierDetection(OutlierDetection):
+    """
+    Remove outliers based on a valid range.
```

Comment:
This doesn't remove anything, it detects (as correctly mentioned in other classes)